fuse_publishers/test/test_pose_2d_publisher.cpp

```diff
@@ -232,48 +251,53 @@ TEST_F(Pose2DPublisherTestFixture, PublishPose)
 TEST_F(Pose2DPublisherTestFixture, PublishPoseWithCovariance)
 {
   // Test that the expected PoseWithCovarianceStamped message is published
+  auto node = rclcpp::Node::make_shared("test_pose_2d_publisher_node");
+  executor_->add_node(node);
+
+  fuse_core::getParam(node, "map_frame", std::string("test_map"));
+  fuse_core::getParam(node, "odom_frame", std::string("test_odom"));
+  fuse_core::getParam(node, "base_frame", std::string("test_base"));
+  fuse_core::getParam(node, "publish_to_tf", false);
 
   // Create a publisher and send it the graph
-  private_node_handle_.setParam("test_publisher/map_frame", "test_map");
-  private_node_handle_.setParam("test_publisher/odom_frame", "test_odom");
-  private_node_handle_.setParam("test_publisher/base_frame", "test_base");
-  private_node_handle_.setParam("test_publisher/publish_to_tf", false);
   fuse_publishers::Pose2DPublisher publisher;
-  publisher.initialize("test_publisher");
+  publisher.initialize(node, "test_publisher", 0);
   publisher.start();
 
   // Subscribe to the "pose_with_covariance" topic
-  ros::Subscriber subscriber = private_node_handle_.subscribe(
-    "test_publisher/pose_with_covariance",
-    1,
-    &Pose2DPublisherTestFixture::poseWithCovarianceCallback,
-    reinterpret_cast<Pose2DPublisherTestFixture*>(this));
+  auto subscriber1 = node->create_subscription<geometry_msgs::msg::PoseWithCovarianceStamped>(
+    "/test_publisher/pose_with_covariance", 1,
+    std::bind(&Pose2DPublisherTestFixture::poseWithCovarianceCallback, this, std::placeholders::_1));
 
   // Send the graph to the Publisher to trigger message publishing
   publisher.notify(transaction_, graph_);
 
   // Verify the subscriber received the expected pose
-  rclcpp::Time timeout = node_->now() + rclcpp::Duration::from_seconds(10.0);
-  while ((!received_pose_with_covariance_msg_) && (node_->now() < timeout))
+  rclcpp::Time timeout = node->now() + rclcpp::Duration::from_seconds(20.0);
+  while ((!received_pose_with_covariance_msg_) && (node->now() < timeout))
   {
-    rclcpp::sleep_for(rclcpp::Duration::from_seconds(0.10));
+    rclcpp::sleep_for(rclcpp::Duration::from_seconds(0.10).to_chrono<std::chrono::nanoseconds>());
   }
 
+  rclcpp::sleep_for(rclcpp::Duration::from_seconds(5).to_chrono<std::chrono::nanoseconds>());
```

Comment:
Why the extra 5 second sleep here?

Reply:
No idea :o

https://github.com/locusrobotics/fuse/compare/f9d587a8c481bab1398f71dd2a4a5eb28fa82547..6ee89e8d6b24b282fb99c14482fe2d6412dcc3e9